pkg/endpointslices/filter.go

```diff
@@ -2,16 +2,53 @@ package endpointslices
 
 import (
 	corev1 "k8s.io/api/core/v1"
+	discoveryv1 "k8s.io/api/discovery/v1"
 )
 
+type EndpointPortInfo struct {
+	EndpointPort  discoveryv1.EndpointPort
+	ContainerPort corev1.ContainerPort
+}
+
+// getEndpointSlicePortsFromPod returns endpoint slice ports that are open in the given pod.
```

Comment:
`that are open in the given pod` is not accurate, we can just say this returns the corresponding slice of epsinfo for the pod